app/src/main/java/ch/admin/bag/dp3t/home/HomeFragment.java

```diff
@@ -355,6 +361,35 @@ private void setupScrollBehavior() {
 		});
 	}
 
+	private void showEndIsolationDialogIfNecessary() {
+		Observer<TracingStatusInterface> observer = new Observer<TracingStatusInterface>() {
+			@Override
+			public void onChanged(TracingStatusInterface tracingStatusInterface) {
+				long isolationEndDialogTimestamp = secureStorage.getIsolationEndDialogTimestamp();
+				if (isolationEndDialogTimestamp != -1L && System.currentTimeMillis() > isolationEndDialogTimestamp) {
+					new AlertDialog.Builder(requireContext(), R.style.NextStep_AlertDialogStyle)
+							.setTitle(R.string.homescreen_isolation_ended_popup_title)
+							.setMessage(R.string.homescreen_isolation_ended_popup_text)
+							.setPositiveButton(R.string.answer_yes, (dialog, which) -> {
+								if (tracingStatusInterface.isReportedAsInfected()) {
+									tracingStatusInterface.resetInfectionStatus(getContext());
+								}
+								secureStorage.setIsolationEndDialogTimestamp(-1L);
+							})
+							.setNegativeButton(R.string.answer_no, (dialog, which) -> {
+								long newTimestamp = isolationEndDialogTimestamp + TimeUnit.DAYS.toMillis(1);
```

Comment:
Shouldn't that be now + 1day? Since `isolationEndDialogTimestamp + TimeUnit.DAYS.toMillis(1)` might be still in the past, triggering the popup again on the next screen view.

Reply:
Makes absolutely sense. Fixed now